binscripts/rvm-installer

```diff
@@ -278,7 +281,8 @@ fetch_versions()
       _url=https://${_domain}/api/v3/repos/${_account}/${_repo}/tags
       ;;
   esac
-  __rvm_curl -sS ${_url} |
+
+  { __rvm_curl -sS "${_url}" || warn "...the preceeding error with code $? occurred while fetching $_url" ; } |
```

Comment:
@pkuczynski I've updated this message to appear like so when a GitHub fetch_versions() fails but a BitBucket succeeds:

<img width="754" alt="screenshot-of-recoverable-fetch-failure" src="https://user-images.githubusercontent.com/208529/61164332-cdda3900-a4d9-11e9-9876-9c58d2d262cc.png">

It has the following characteristics:
* Only emits log output when something bad (recoverable) happens.
* Emits to STDERR, to avoid breaking the STDOUT capture of _version=$(fetch_version ...)

I believe this honors spirit of what we discussed in https://github.com/rvm/rvm/pull/4728#discussion_r302555052

However, it deviates slightly from your original suggestion, which involved deferring the display of the curl STDERR ({error status curl}) output until after most of the log message:
> Ideally, it should print Fetching https://api.github.com/repos/rvm/rvm/tags has failed (403 Forbidden) when curl failed, so basically Fetching {url} ({error status curl})

----

If desired, I do have [this alternative version](https://github.com/sfgeorge/rvm/compare/improve-rvm-installer-error-handling...sfgeorge:improve-rvm-installer-error-handling.defer-stderr) which does honor your request and emits the curl STDERR message last, by first capturing it and then displaying it.  However, it proves quite difficult to capture both STDOUT and STDERR separately in bash without mushing them together into one stream.  [Most suggests out there](https://stackoverflow.com/questions/11027679/capture-stdout-and-stderr-into-different-variables) involve some use of eval().  More reasonable suggestions involve capturing to named-pipes, or to simple files.  My alternate approach opts for the latter for simplicity - dumping the STDERR output to a file, and emitting that if a non-zero exit code is detected.  I'm keeping this approach out of the PR for now since it has a few potential draw-backs:
* Adding an additional file to write to the user's disk seems less than ideal.
* If the capturing & outputting mechanism I've added fails for whatever reason (such as disk full or permission denied), then curl's STDERR will have been captured but never displayed.

Let me know your thoughts on a preferred approach.

Reply:
I think it's not worth to go so far as in the alternative PR. What you did here should be good enough - even if it deviates from the nicest output it should be helpful enough I think.

Reply:
@pkuczynski Thanks very much, man.  I appreciate it!

On an unrelated note, I think the [merge into master unfortunately broke](https://travis-ci.org/rvm/rvm/builds/558900224) the signature for rvm-installer; and someone has already noticed (#4736).  Would you mind re-signing to make it all happy?

Reply:
@pkuczynski or @mpapis Can you update the signature for rvm-installer.asc?  This is unfortunately affecting some builds (#4736).